### PR TITLE
fix(#1444): improve sidemenu url matching

### PR DESCRIPTION
**Pull request by chrisolsen**

### CODE REVIEW

File: libs/web-components/src/components/side-menu-group/SideMenuGroup.svelte [hunk 1]
```diff
@@ -71,36 +71,41 @@
...
-        child.classList.add("current");
-        notifyParent(true);
-      } else {
-        child.classList.remove("current")
+      const weight = isUrlMatch(document.location, url);
+      if (weight > maxMatchWeight) {
+         maxMatchWeight = weight;
+          matchedChild = child; 
       }
+      child.classList.remove("current")
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/web-components/src/components/side-menu-group/SideMenuGroup.svelte:96]: Based on the logic used to set which is the current url it only ever finds one url which is current, so I am not sure how it is setting them all to current on your machine. When I use the these updates in the docs site only the top most link stays highlighted (which is not right), but they all aren't highlighted.

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**